lib/ansible/runner/connection_plugins/winrm.py

```diff
@@ -72,11 +84,10 @@ def _winrm_connect(self):
         if cache_key in _winrm_cache:
             vvvv('WINRM REUSE EXISTING CONNECTION: %s' % cache_key, host=self.host)
             return _winrm_cache[cache_key]
-        transport_schemes = [('plaintext', 'https'), ('plaintext', 'http')] # FIXME: ssl/kerberos
-        if port == 5985:
-            transport_schemes = reversed(transport_schemes)
         exc = None
-        for transport, scheme in transport_schemes:
+        for transport, scheme in self.transport_schemes['http' if port == 5985 else 'https']:
+            if transport == 'kerberos' and not HAVE_KERBEROS:
+                continue
```

Comment:
The thinking behind not doing that earlier was that it would retain backward compatibility with existing installations that don't have kerberos.py installed. With that change, since the `kerberos` is in both transport schemes and will always be tried, all Windows admins would start getting exceptions until they install kerberos.